task10/src/Basics.hs

```diff
@@ -9,39 +9,51 @@ import Prelude hiding (head, tail, take, drop, filter, foldl, concat, (++))
 
 -- 1. head' возвращает первый элемент непустого списка
 head' :: [a] -> a
-head' = undefined
+head' (x:_) = x
 
 -- 2. tail' возвращает список без первого элемента, для пустого - пустой
 tail' :: [a] -> [a]
-tail' = undefined
+tail' [] = []
+tail' (_:xs) = xs
 
 -- 3. take' возвращает первые n >= 0 элементов исходного списка
 take' :: Int -> [a] -> [a]
-take' = undefined
+take' 0 _ = []
+take' n (x:xs)  = x : take' (n - 1) xs
 
 -- 4. drop' возвращает список без первых n >= 0 элементов; если n больше длины
 -- списка, то пустой список.
 drop' :: Int -> [a] -> [a]
-drop' = undefined
+drop' 0 x = x
+drop' _ [] = []
+drop' n (_:xs) = drop' (n - 1) xs
 
 -- 5. filter' возвращает список из элементов, для которых f возвращает True
 filter' :: (a -> Bool) -> [a] -> [a]
-filter' f xs = undefined
+filter' _ [] = []
+filter' f (x:xs) | f x       = x : filter' f xs
```

Comment:
Вкусовщина, но я бы перенес на следующую строку и табами бы выровнял

Reply:
что конкретно перенести?

Reply:
А
гарды, забыл сказать, извините

Reply:
все еще не понял что требуется перенести? гарды в смысле так
`filter' f (x:xs)`
                 `| f x       = x : filter' f xs`
                 `| otherwise = filter' f xs`
? (с выравниванием)

Reply:
Да